ballerina-command/src/main/java/org/ballerinalang/command/cmd/PullCommand.java

```diff
@@ -49,15 +49,30 @@ public void execute() {
             return;
         }
 
-        if (pullCommands == null) {
+        if (pullCommands == null || pullCommands.size() == 0) {
             throw ErrorUtil.createUsageExceptionWithHelp("distribution is not provided");
-        } else if (pullCommands.size() == 1) {
-            ToolUtil.handleInstallDirPermission();
-            ToolUtil.downloadDistribution(getPrintStream(), pullCommands.get(0), false);
-            ToolUtil.use(getPrintStream(), pullCommands.get(0));
-        } else if (pullCommands.size() > 1) {
+        }
+
+        if (pullCommands.size() > 1) {
             throw ErrorUtil.createUsageExceptionWithHelp("too many arguments given");
         }
+        ToolUtil.handleInstallDirPermission();
+        PrintStream printStream = getPrintStream();
+        String distribution = pullCommands.get(0);
+        String distributionType = distribution.split("-")[0];
```

Comment:
What if the dist name has completely wrong name? Have we handled that? Looks like this line will break in that case.

Reply:
You can get "" still split will not fail. null values are handled in upper code.